go.mod

```diff
@@ -52,6 +52,7 @@ require (
 	github.com/ajg/form v1.5.1 // indirect
 	github.com/andybalholm/brotli v1.0.5 // indirect
 	github.com/apapsch/go-jsonmerge/v2 v2.0.0 // indirect
+	github.com/avast/retry-go/v4 v4.5.0
```

Comment:
I would have preferred using https://github.com/cenkalti/backoff as I know that better.

Reply:
Discussed with DM, giving a try to https://github.com/avast/retry-go